src/Game.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Game.h"
+
+Game::Game()
+{
+    initscr();
+    noecho();
+    int row, col;
+    getmaxyx(stdscr, row, col);
+    move(row/2, col/2);
+    timeout(setting::speed * 100);
+}
+
+void Game::startGame()
+{
+    spawnSnake();
+    spawnFood();
+
+    while (!m_snake.isOver())
+    {
+        displayState();
+        updateState();
+
+        if(m_snake.isHitItself())
+           break;
+    }
+}
+
+void Game::spawnSnake()
+{
+    for(const auto& e: m_snake.getPos())
+        m_board.setCellValue(e, Board::snake);
+}
+
+void Game::spawnFood()
+{
+    m_board.setCellValue(m_food.getPos(), Board::food);
+}
+
+Snake::Direction Game::getInput() {
+    char input{};
+    if ((input = getch()) != ERR) {
+        switch (input) {
+            case 'w':
+                return Snake::Direction::left;
+            case 's':
+                return Snake::Direction::right;
+            case 'a':
+                return Snake::Direction::up;
+            case 'd':
+                return Snake::Direction::down;
+            default: ;
+        }
+    }
+    return Snake::Direction::max_directions;
```

Comment:
move this to default case in switch

Reply:
this line should be after if ((input = wgetch(m_winGame)) != ERR), since it serves to set the previous direction to the current